modules/endpoint-group/main.tf

```
@@ -1,5 +1,20 @@
 locals {
-  enabled = module.this.enabled
+  enabled                 = module.this.enabled
+  endpoint_configurations = try(var.config.endpoint_configuration, toset([]))
```

nit: could the `toset()` be omitted?

```suggestion
  endpoint_configurations = try(var.config.endpoint_configuration, [])
```

---

Ah yes, good catch

---

BTW, this
```
try(var.config.endpoint_configuration, [])
```
 will not work correctly in all cases.

it "throw" only if `endpoint_configuration` is not set at all in `var.config`.
but if it set to "" or to `null`, it will not "throw" and use it.
If we don't want to allow "" or `null` here (and since it's a list, we prob don't') , we need to do something like this:

```
endpoint_configurations = try(var.config.endpoint_configuration, null) != null ? var.config.endpoint_configuration : []
```

---

Resolved